docs: v3 Tutorial (#30843)
gatsbyjs/gatsby: docs/docs/tutorial/part-0/index.mdx

---
title: Set Up Your Development Environment
tableOfContentsDepth: 2
---

import { Announcement } from 'gatsby-interface'

## Introduction

Welcome to the Gatsby Tutorial! We're excited you're here.

Before you start building your first Gatsby site, you’ll need to familiarize yourself with some core web technologies and make sure that you have installed all required software tools.

By the end of this part of the Tutorial, you will:

* Have a basic understanding of the key technologies needed to create a Gatsby site.
* Install all the required software tools.
* Create all the necessary accounts for deploying your site online.

## Background Knowledge

> The Tutorial is intended to be approachable as approachable as possible for people who are new to Gatsby and React. With that said, if you are brand new to web development (welcome!), you might find it helpful to learn some fundamentals first.

Reviewer: ```suggestion
The Tutorial is intended to be as approachable as possible for people who are new to Gatsby and React. With that said, if you are brand new to web development (welcome!), you might find it helpful to learn some fundamentals first.
```

        * [Command line instructions for Windows](https://www.lifewire.com/how-to-open-command-prompt-2618089)
        * [Command line instructions for Linux](https://www.howtogeek.com/140679/beginner-geek-how-to-start-using-the-linux-terminal/).

In addition to the ones listed above, Gatsby uses a few more technologies under the hood. The Tutorial will teach you everything you need to know about the following tools. But if you're interested in getting a head start, here's a bit more information:

* **React**: A code library (built with JavaScript) for building user interfaces. It’s the framework that Gatsby uses to build pages and structure content.
    * Learn more about React with the [React course on freeCodeCamp](https://www.freecodecamp.org/learn/front-end-libraries/#react).
* **GraphQL**: A query language that allows you to pull data into your website. It’s the interface that Gatsby uses for managing site data.
    * Learn more about GraphQL with the [How to GraphQL tutorial](https://www.howtographql.com/).

## Installation Guide

The rest of this part of the Tutorial walks you through how to install the following tools:

> * [Node.js](#nodejs) (v12 or newer)

Reviewer: ```suggestion
* [Node.js](#nodejs) (v12.13 or newer)
```

* [Git](#git)
* [Gatsby command line interface (CLI)](#gatsby-cli) (v3 or newer)
* [Visual Studio Code](#vs-code)

### Node.js

> [Node.js](https://nodejs.dev/learn) is an environment that can run JavaScript code outside of a web browser. Gatsby is built with Node.js. To get up and running with Gatsby, you’ll need to have Node.js version 12 (or newer) installed on your computer.

Reviewer: ```suggestion
[Node.js](https://nodejs.dev/learn) is an environment that can run JavaScript code outside of a web browser. Gatsby is built with Node.js. To get up and running with Gatsby, you’ll need to have Node.js version 12.13 (or newer) installed on your computer.
```

To be _super_ correct on this one

Author: Love it! Thanks :)

<details>

<summary>

#### Mac instructions

</summary>

To install Gatsby and Node.js on a Mac, it is recommended to use [Homebrew](https://brew.sh/). A little set-up in the beginning can save you from some headaches later on!

##### How to install or verify Homebrew on your computer:

1. Open the Terminal app.
2. See if Homebrew is installed by typing the command below. You should see "Homebrew" and a version number.

```shell
brew -v
```

3. If not, [download and install Homebrew](https://docs.brew.sh/Installation).
4. Once you've installed Homebrew, repeat step 2 to verify that you get the expected version message.

##### Install Xcode Command Line Tools:

1. Open your Terminal.
2. Install Xcode Command line tools by running:

```shell
xcode-select --install
```

<Announcement>
If that fails, download it [directly from Apple's site](https://developer.apple.com/download/more/), after signing-in with an Apple developer account.
</Announcement>

3. After being prompted to start the installation, you'll be prompted again to accept a software license for the tools to download.

##### Install Node

1. Open your Terminal

> 2. Install node with Homebrew:

Reviewer 2: ```suggestion
2. Install Node with Homebrew:
```

```shell
nvm --version
```

5. Continue with the section: [Set default Node.js version](#set-default-nodejs-version)

##### Arch, Manjaro and other `pacman` based distros:

1. Make sure your distribution is ready to go:

```shell
sudo pacman -Syu
```

2. These distros come installed with curl, so you can use that to download nvm:

```shell
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.1/install.sh | bash
```

3. Before using nvm, you need to install additional dependencies by running:

```shell
sudo pacman -S grep awk tar
```

4. Confirm this has worked. The output should be a version number.

```shell
nvm --version
```

5. Continue with the section: [Set default Node.js version](#set-default-nodejs-version)

##### Fedora, RedHat, and other `dnf` based distros:

1. These distros come installed with curl, so you can use that to download nvm:

```shell
curl -o- https://raw.githubusercontent.com/nvm-sh/nvm/v0.35.1/install.sh | bash
```

2. Confirm this has worked. The output should be a version number.

```shell
nvm --version
```

3. Continue with the section: [Set default Node.js version](#set-default-nodejs-version)

##### Set default Node.js version

> When nvm is installed, it does not default to a particular node version. You’ll need to install the version you want and give nvm instructions to use it. This example uses the version 10 release, but more recent version numbers can be used instead.

Reviewer 2: We should use Node 14 in the examples, as it's the LTS release.

> When nvm is installed, it does not default to a particular node version. You’ll need to install the version you want and give nvm instructions to use it. This example uses the version 10 release, but more recent version numbers can be used instead.
> 
> ```shell
> nvm install 10
> nvm use 10

Reviewer: ```suggestion
When nvm is installed, it does not default to a particular node version. You’ll need to install the version you want and give nvm instructions to use it. This example uses the version 14 release, but more recent version numbers can be used instead.

```shell
nvm install 14
nvm use 14
```

We should tell people to use the current LTS (not something that is obsolete soon)
